### PR TITLE
Bump snakeyaml to 2.0

### PR DESCRIPTION
**Pull request by tmfset**
Issue, https://github.com/KarelCemus/play-i18n/issues/26

Bumping snakeyaml to version 2.0 so that play-i18n plays nicer with other libs.

I haven't built this locally so it's possible there may be issues with the new version.

### CODE REVIEW

**Comment by tmfset** [comment]
Got the build working with some updates to the sbt plugins, there is a code change required for snakeyaml, so working on it.

**Comment by tmfset** [comment]
Tests run locally, all tests pass:
```
➜  play-i18n git:(tf/i26-snakeyaml-2.0) sbt test
[info] welcome to sbt 1.6.1 (Amazon.com Inc. Java 1.8.0_382)
[info] loading settings for project play-i18n-build from plugins.sbt,sonatype.sbt ...
[info] loading project definition from /Users/tom.fiset/src/play-i18n/project
[info] loading settings for project play-i18n from build.sbt,version.sbt ...
[info] set current project to play-i18n (in build file:/Users/tom.fiset/src/play-i18n/)
[info] compiling 9 Scala sources to /Users/tom.fiset/src/play-i18n/target/scala-2.13/classes ...
[info] Non-compiled module 'compiler-bridge_2.13' for Scala 2.13.8. Compiling...
[info]   Compilation completed in 6.374s.
[warn] /Users/tom.fiset/src/play-i18n/src/main/scala/play/ext/i18n/MessageFile.scala:15:55: object JavaConverters in package collection is deprecated (since 2.13.0): Use `scala.jdk.CollectionConverters` instead
[warn]   private def resources = env.classLoader.getResources(joinPaths(messagesPrefix, name)).asScala.filterNot(Resources.isDirectory(env.classLoader, _))
[warn]                                                       ^
[warn] /Users/tom.fiset/src/play-i18n/src/main/scala/play/ext/i18n/loaders/PropertyFileLoader.scala:16:49: method right in class Either is deprecated (since 2.13.0): Either is now right-biased, use methods directly on Either
[warn]     new MessagesParser(messageSource, "").parse.right.map { messages =>
[warn]                                                 ^
[warn] /Users/tom.fiset/src/play-i18n/src/main/scala/play/ext/i18n/loaders/YamlFileLoader.scala:38:61: object JavaConverters in package collection is deprecated (since 2.13.0): Use `scala.jdk.CollectionConverters` instead
[warn]   private def flatten(data: JavaMap): Map[String, String] = data.asScala.collect {
[warn]                                                             ^
[warn] three warnings found
[info] compiling 2 Scala sources to /Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes ...
[warn] /Users/tom.fiset/src/play-i18n/src/test/scala/play/ext/i18n/loaders/YamlFileLoaderSpecs.scala:22:84: object JavaConverters in package collection is deprecated (since 2.13.0): Use `scala.jdk.CollectionConverters` instead
[warn]       val resource = new UrlMessageSource(this.getClass.getClassLoader.getResources("messages.yaml.cs").asScala.toList.head)
[warn]                                                                                    ^
[warn] one warning found
[info] YamlFileLoaderSpecs
[info] YAML parser should
[info]   + handle empty message file
[info]   + handle non-empty message file
[info] Total for specification YamlFileLoaderSpecs
[info] Finished in 111 ms
[info] 2 examples, 0 failure, 0 error
15:58:06.114 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages'.
15:58:06.133 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'jar:file:/Users/tom.fiset/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/typesafe/play/play_2.13/2.8.13/play_2.13-2.8.13.jar!/messages.default'.
15:58:06.150 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/test'.
15:58:06.152 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.en'.
15:58:06.153 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.cs'.
15:58:06.154 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.yaml'.
15:58:06.160 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.yaml.en'.
15:58:06.161 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.en.yaml'.
15:58:06.162 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.yaml.cs'.
15:58:06.163 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/messages.cs.yaml'.
15:58:06.165 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/test_js.json'.
15:58:06.168 [pool-5-thread-1] DEBUG play.api.i18n - Localization file 'file:/Users/tom.fiset/src/play-i18n/target/scala-2.13/test-classes/test_js.json.cs'.
15:58:06.170 [pool-5-thread-1] WARN play.api.i18n - Localization key 'collision' is defined in multiple files for language 'default'.
[info] MultiFormatMessagingPluginSpec
[info] Plugin should
[info]   + read property files
[info]   + read JSON files
[info]   + read YAML files
[info]   + implement advanced YAML features: hierarchy
[info]   + implement advanced YAML features: multi-line strings
[info]   + implement advanced YAML features: references
[info]   + read multiple files with different names
[info]   + collision resolving
[info]   + preserve spaces within quotes
[info] Total for specification MultiFormatMessagingPluginSpec
[info] Finished in 60 ms
[info] 9 examples, 0 failure, 0 error
[info] Passed: Total 11, Failed 0, Errors 0, Passed 11
[success] Total time: 11 s, completed Sep 11, 2023 3:58:06 PM
```

File: src/main/scala/play/ext/i18n/loaders/YamlFileLoader.scala [hunk 1]
```diff
@@ -1,18 +1,14 @@
 package play.ext.i18n.loaders
 
-import play.api.PlayException
-import play.api.PlayException.ExceptionSource
-import play.api.i18n.Messages.MessageSource
-
-import play.ext.i18n.MessagesLoader
-
-import org.yaml.snakeyaml.constructor.Constructor
+import org.yaml.snakeyaml.Yaml
```
Comment (tmfset) [inline src/main/scala/play/ext/i18n/loaders/YamlFileLoader.scala:3]: I optimized the imports here using intellij automatically, happy to revert though if this is too noisy.

File: src/main/scala/play/ext/i18n/loaders/YamlFileLoader.scala [hunk 2]
```diff
@@ -27,7 +23,7 @@ class YamlFileLoader extends MessagesLoader {
   override def apply(messageSource: MessageSource, messageSourceName: String): Either[ExceptionSource, Map[String, String]] =
     try {
       // YAML document parser
-      val yaml = new Yaml(new Constructor(), new Representer(), new DumperOptions(), new CustomResolver())
+      val yaml = new Yaml()
```
Comment (tmfset) [inline src/main/scala/play/ext/i18n/loaders/YamlFileLoader.scala:26]: The default `Yaml` constructor looks the most like what we're trying to do here, just fill in all the defaults.

**Comment by tmfset** [comment]
@KarelCemus ready for review when you get time to take a look 🙏

**Review by KarelCemus — APPROVED** [review]
Looking good, thanks for your contribution

**Pull request merged**